### PR TITLE
fix:(DB\Smart_Scripts):WateryInvaders should not despawn

### PR DESCRIPTION
**Pull request by Ragelezz**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7418
- Closes https://github.com/chromiecraft/chromiecraft/issues/1455

## Tests Performed:

- .go c id 14458 and it despawned either after 500ms (when out of combat) either 2 mins after spawned.
- 

## How to Test the Changes:

1. .go c id 14458 and look for the portal, stay near portal and look at elementals, they won't despawn anymore
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1632569421493685900.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+UPDATE `smart_scripts` SET `action_param3`=990000 WHERE `entryorguid`=179665 AND `source_type`=1 AND `id`=0 AND `link`=0;
+
+
+DELETE FROM `smart_scripts` WHERE `entryorguid`=14458;
+
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
+
+(14458, 0, 0, 0, 54, 0, 100, 0, 0, 0, 0, 0, 0, 89, 11, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Watery Invader - Is Summoned - Random Move'),
+
+(14458, 0, 1, 0, 0, 0, 100, 0, 0, 7000, 8000, 14000, 0, 11, 19133, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Watery Invader - In Combat - Cast Frost Shock');
```
Comment (Helias) [inline data/sql/updates/pending_db_world/rev_1632569421493685900.sql:13]: these lines are not necessary
```suggestion
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(14458, 0, 0, 0, 54, 0, 100, 0, 0, 0, 0, 0, 0, 89, 11, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Watery Invader - Is Summoned - Random Move'),
(14458, 0, 1, 0, 0, 0, 100, 0, 0, 7000, 8000, 14000, 0, 11, 19133, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Watery Invader - In Combat - Cast Frost Shock');
```

Comment (Ragelezz) [reply data/sql/updates/pending_db_world/rev_1632569421493685900.sql:13]: lol on my file is not like this, strange

**Comment by syssneck** [comment]
Tested, now they don't despawn, but the event spawns new ones now and then (I'm at 5 now):
grafik

File: data/sql/updates/pending_db_world/rev_1632568882031573700.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1632568882031573700');
+
+UPDATE `smart_scripts` SET `action_param3`='990000' WHERE  `entryorguid`=179664 AND `source_type`=1 AND `id`=0 AND `link`=0;
+
+DELETE FROM `smart_scripts` WHERE `entryorguid`=14462;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
+(14462, 0, 0, 0, 54, 0, 100, 0, 0, 0, 0, 0, 0, 89, 11, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Thundering Invader - Is Summoned - Random Move'),
+(14462, 0, 1, 0, 0, 0, 100, 0, 0, 7000, 8000, 14000, 0, 11, 23114, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Thundering Invader - In Combat - Cast Earth Shock');
```
Comment (syssneck) [inline data/sql/updates/pending_db_world/rev_1632568882031573700.sql:8]: This whole part is about Thundering Invaders and has nothing to do with the Watery Invaders in Winterspring, right?

Comment (Ragelezz) [reply data/sql/updates/pending_db_world/rev_1632568882031573700.sql:8]: Yea, merged 2 files into one by mistake, all good from the initial commit.

**Comment by Ragelezz** [comment]
Well.. they will continue to spawn until you kill their leader of that type but I don't think any normal player would stay hours to have 1000 npcs spawned, right ?
https://classic.wowhead.com/guides/elemental-invasions-wow-classic

**Comment by syssneck** [comment]
Everyone would farm them without killing their leader:
grafik

**Comment by Ragelezz** [comment]
Some players will opt to delay in killing the named leader as to farm materials such  Essence of Fire,  Essence of Earth,  Essence of Water and  Essence of Air, as all of the invaders have a chance to drop them. Additionally, they can drop some of the lesser quality elemental items such as:
 Elemental Fire &  Heart of Fire
 Elemental Earth &  Core of Earth
 Elemental Water &  Globe of Water
 Elemental Air &  Breath of Wind

???

**Comment by Ragelezz** [comment]
@nolt  WIP for ? No one replied to what I said with something ;-?

**Pull request closed without merging**